### ISSUE TITLE
Various axe accessibility checks failing for ContextMenu

### ISSUE THREAD

**Issue by jpveooys** [Royal-Navy/design-system#3403]
There are various axe accessibility checks failing for the ContextMenu component. These relate to missing ARIA roles and possibility incorrect use of `div`s.

These aren't being tested currently, as the stories don't have the context menu open by default.

Related: #2570, #1928
